lib/ansible/modules/windows/win_timezone.ps1

```diff
@@ -19,28 +19,32 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
-$params = Parse-Args $args;
-
-$result = New-Object psobject @{
-    win_timezone = New-Object psobject
+$result = @{
     changed = $false
+    win_timezone = @{}
 }
 
-$timezone = Get-Attr -obj $params -name timezone -failifempty $true -resultobj $result
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$timezone = Get-AnsibleParam -obj $params -name "timezone" -failifempty $true -resultobj $result
 
 Try {
     # Get the current timezone set
     $currentTZ = $(tzutil.exe /g)
-    If ($LASTEXITCODE -ne 0) { Throw "An error occurred when getting the current machine's timezone setting." /
+    If ($LASTEXITCODE -ne 0) {
+        Throw "An error occurred when getting the current machine's timezone setting."
+    }
 
     If ( $currentTZ -eq $timezone ) {
         Exit-Json $result "$timezone is already set on this machine"
```

Comment:
Do we need this here or should we just remove this if branch and just exit at the end of the code like normal